### ISSUE TITLE
Search is broken on branches such as `stable`

### ISSUE THREAD

**Issue by davidak** [godotengine/godot-docs#10062]
**Issue description:**

I don't get any results when using the search.

![Screenshot from 2024-10-08 15 09 01](https://github.com/user-attachments/assets/5aff5965-295e-47c0-a159-823336f2caf6)

**URL to the documentation page:** e.g. https://docs.godotengine.org/en/stable/search.html?q=export&check_keywords=yes&area=default#

It works for the docs of version:

- latest
- 4.1
- 3.6
- 3.4
- 3.3
- 3.2
- 2.1

e.g. https://docs.godotengine.org/en/3.6/search.html?q=export&check_keywords=yes&area=default# works

I have tested the docs for every version in the default language (en). The german (de) docs all work.

Reproducible in all tested browsers:

- Google Chrome 128.0.6613.137
- epiphany-46.1 (aka GNOME Web, with elementary patches)

**Comment by Yarwin** [comment]
doesn't work on **Firefox 130.0.1 (64-bit)** either.

**Comment by Sundwell** [comment]
Doesn't work on macos/chrome either

**Comment by Supergeek** [comment]
Clear your cache/cookies for the site. Fixed it for me.

**Comment by rishitkhare** [comment]
> Clear your cache/cookies for the site. Fixed it for me.

This did not fix the issue for me.

**Comment by TAGames** [comment]
> > Clear your cache/cookies for the site. Fixed it for me.
> 
> This did not fix the issue for me. I am on Safari. I think it's an issue with the site/API itself since the problem appears on a variety of browsers.
> 
> The issue does not appear when downloading the offline version (10/08/2024) of the documentation.

The offline version shows me the title where it was found, but didn't the search results show more infos?

**Comment by dannroda** [comment]
Doesn't work on zen-browser 1.0.1-a.7 (firefox 130.0) and chromium (129.0.6668.100) under linux
Clearing cache/cookies didn't solve the problem. Chromium is a fresh install.

**Comment by rishitkhare** [comment]
> > > Clear your cache/cookies for the site. Fixed it for me.
> > 
> > 
> > This did not fix the issue for me. I am on Safari. I think it's an issue with the site/API itself since the problem appears on a variety of browsers.
> > The issue does not appear when downloading the offline version (10/08/2024) of the documentation.
> 
> The offline version shows me the title where it was found, but didn't the search results show more infos?

This also happened to me, I just forgot to mention it.

**Comment by botteu** [comment]
Yes. Seach doesn't yield any results in the stable version of the documentation at the moment. I'm using Chrome on MacOs and Windows 11.

**Comment by mifuyne** [comment]
> > > Clear your cache/cookies for the site. Fixed it for me.
> > 
> > 
> > This did not fix the issue for me. I am on Safari. I think it's an issue with the site/API itself since the problem appears on a variety of browsers.
> > The issue does not appear when downloading the offline version (10/08/2024) of the documentation.
> 
> The offline version shows me the title where it was found, but didn't the search results show more infos?

If you run a local webserver, then search will display more info than just the title. You should be able to use something like Python's `http.server` module. I'm using darkhttpd on Termux on Android. Hope that helps until the stable docs search function works again!

> Yes. Seach doesn't yield any results in the stable version of the documentation at the moment. I'm using Chrome on MacOs and Windows 11.
> 
> Search seems to be working on latest (not stable) version of the docs though.

I'm seeing this as well. Search works on latest but not stable. Tested with Firefox Beta on Android and Firefox on Windows 10.

**Comment by mmailloux22** [comment]
The search is not working on the Brave
 Browser either.
Version 1.70.126 Chromium: 129.0.6668.100 (Official Build) (64-bit)

And clearing my cache did not resolve the issue.

**Comment by TAGames** [comment]
> Yes. Seach doesn't yield any results in the stable version of the documentation at the moment. I'm using Chrome on MacOs and Windows 11.
> 
> Search seems to be working on latest (not stable) version of the docs though.

Yes, latest works for me too. Thx, good to know!

**Comment by bloemer** [comment]
A good friend noticed, that the search triggered by the "other" input field works.

Here is an screenshot from the docs.

![Screenshot_20241013-131500964 (1)](https://github.com/user-attachments/assets/69c2a923-4371-444b-bb34-212fae5cfd0e)

The regular search inside the side menu does not work, but the one on the "bottom"

**Comment by Calinou** [comment]
Can anyone affected by this issue reproduce this in a private browsing tab on a fresh browser profile (or at least with all add-ons disabled)? If so, try reproducing this on a different device on the same network.

I'm asking since I can't reproduce this on `/en/stable/` with [Cromite](https://www.cromite.org) 129 on Android 14.

**Comment by anniryynanen** [comment]
I had this issue on Saturday but I can't reproduce it anymore.

**Comment by syntaxerror247** [comment]
i can not reproduce this issue on my mobile with chrome Android 14, But search is broken on my windows pc for stable branch and i cannot find any errors in console either.

**Comment by TAGames** [comment]
> I had this issue on Saturday but I can't reproduce it anymore.

Same. Today it works.

**Comment by dannroda** [comment]
For me now works in any version BUT stable.
Its odd that it works by using 4.3 and not stable.

**Comment by mhilbrunner** [comment]
Thanks everyone for reporting and providing details (please keep doing so if you are affected, and also if the issue disappears for you!). We're still investigating and this continues to be hard to pin down (i.e., for me search works on all branches, in various browsers, both desktop and mobile).

**Comment by syntaxerror247** [comment]
I just checked and it is now working for me 🎉. 
It seems like `ReadTheDocs` is updated because its floating button looks different now.

**Comment by SweatyTryhardx10** [comment]
The search bar (in the top left) has not worked in the past few days for me. As noted by @bloemer, the search bar within the 'Read the Docs' floating menu _does_ work.

I've tried Firefox v131.0.3 and Chrome v129.0.6668.101 - both don't work for me.

I looked in the Networking tab of the dev tools after a hard reload and there are no errors shown i.e. no content is missed or blocked.

**Comment by daveTheOldCoder** [comment]
I'm experiencing the same problems posted in this issue.

Is the search page driven completely by the code in this repository, or does it use a third-party service?

If it only uses the code in this repository, were any relevant changes made shortly before this issue was opened?

**Comment by Jack-023** [comment]
This issue seemed to be resolved for the last couple of weeks but it seems to have returned in the last ~1 hour. I tried searching for "Node" on the stable version of the docs and got no results in a completely fresh Firefox profile.
Browser version: `131.0.3 (64-bit)`

**Comment by mhilbrunner** [comment]
> Is the search page driven completely by the code in this repository, or does it use a third-party service?

The search is provided by ReadTheDocs, not us.

**Comment by bloemer** [comment]
> The search is provided by ReadTheDocs, not us.

Interestingly, on their page –
[Read the docs](https://docs.readthedocs.io/en/stable/) – when I'm "using" the search bar inside the menu on the left hand side, it opens up the same pop up, as if I would click on the search input field inside the _floating search button_.

Looks like a workaround, or new standard. Maybe an outdated version of the CMS?

**Comment by Ruruie-FGSC** [comment]
I discover this amidst gamejams so please understand when I say I don't have time to do the testing I wish I did. >_<

Search worked normally on /stable/ on both an auto-updated and presumably up to date Firefox running just ublock origin as well as mildly out of date Librewolf running ublock origin and various other things during LD56.

Search stopped working normally on /stable/ on all permutations I've tried on fully updated versions of Firefox and Librewolf regardless of things being turned on or off, cookies and data being cleared, etc. I never fully cleared _all_ data or cookies, but I couldn't find anything obvious left (ie things from godot or read the docs).

Currently, search works perfectly on /4.3/ as well as offline docs.

A friend using Firefox has 0 issues with any kind of searching.

**Comment by mhilbrunner** [comment]
Hello everyone 👋 Thanks again for all of your feedback.

1. Can anyone still reproduce this on the `stable` version of the docs?
2. Can anyone still reproduce this on the `latest` version of the docs?

We just did some updates on `latest`, and RTD may also have improved things on their side, so knowing whether any of that helped would be useful.

**Comment by daveTheOldCoder** [comment]
`stable` version - Search works normally.

`latest` version - When I type in the search box in the left pane, a popup appears for typing in the search string. That works, except that the text I type is invisible, which makes it difficult to use. Maybe the dark system theme I'm using is causing the text to be invisible?

**Comment by mhilbrunner** [comment]
> That works, except that the text I type is invisible

Yeah, it is white text on a white background, I can reproduce that. Seems there is some CSS issue with the RTD `search.css`, not sure if thats on their end. Will investigate.

**Comment by dannroda** [comment]
`latest`and `stable` now works perfectly for me

**Comment by mifuyne** [comment]
Both `latest` and `stable` are working on my end. I also checked `4.3` and `3.6` and they're working fine too!

**Comment by paskausks** [comment]
Checked both `latest` and `stable`, both work now.

**Comment by Jack-023** [comment]
I just tried stable. Not working.

**Comment by bloemer** [comment]
Now that it's working for some people and some doesn't, maybe it's not an issue with the environment, but with services in the background? Me for example, I use a piHole. I need to check if one of the necessary connections are listed there.

**Comment by Jack-023** [comment]
I just tried searching with blocking disabled on my pihole and a clean firefox profile. Still no results on stable.

**Comment by mhilbrunner** [comment]
@Jack-023 But `latest` works for you? We may take a chance and backport the updates we did to latest then and see if that fixes it. Seems no one has any issues with `latest` anymore, so far.

**Comment by Jack-023** [comment]
> @Jack-023 But `latest` works for you?

Yes, latest has worked this entire time for me.

**Comment by daveTheOldCoder** [comment]
Maybe this should be in a new issue, but in my opinion, "stable" and "latest" are not useful because they change when new versions are released. I prefer to use the explicit version, e.g. 4.3, that corresponds to the Godot version I'm using. If you answer someone's question by posting a documentation link that uses "stable", the link may be out of date when someone reads the post at the later date.

I only use "latest" because that's the only way of accessing the unreleased version that's currently in development.

**Comment by AThousandShips** [comment]
That's unrelated, please open a new issue for that and keep this on topic

**Comment by Jack-023** [comment]
Search appears to be working again.

**Comment by daveTheOldCoder** [comment]
For me, it's working for 3.6, 4.0, 4.1, 4.2, 4.3, latest and stable.
It's not working for 3.5.

**Comment by bloemer** [comment]
Can approve. Works for me too.

**Comment by Jack-023** [comment]
Search on stable is broken again.

**Comment by tetrapod00** [comment]
Anecdotally, this seems to be resolved for a couple months, so I'm closing the issue. **If anyone encounters broken search again, please comment and we can reopen and look into this again.**
